### PR TITLE
Use passed clientId instead of default constant value.

### PR DESCRIPTION
**Pull request by zeroadam**
Updating the client class to accept the clientId when passed, instead of always using the default constant.

### CODE REVIEW

File: src/IdentityClient.cs [hunk 1]
```diff
@@ -45,6 +45,8 @@ public class IdentityClient
         /// </summary>
         private readonly string identityClientSecret;
 
+        private readonly string clientId;
```
Comment (vasont) [inline src/IdentityClient.cs:48]: Missing comment header

File: src/IdentityClient.cs [hunk 2]
```diff
@@ -61,11 +63,12 @@ public class IdentityClient
         /// <param name="resourceUri">Contains the URI to the Identity API endpoint.</param>
         /// <param name="identityClientSecret">Contains the API resource client secret.</param>
         /// <param name="useDiscoveryForEndpoint">Contains a value indicating whether the discovery endpoint is used for token endpoint information.</param>
-        public IdentityClient(Uri authorityUri, Uri resourceUri, string identityClientSecret, bool useDiscoveryForEndpoint = true)
+        public IdentityClient(Uri authorityUri, Uri resourceUri, string identityClientSecret, string clientId, bool useDiscoveryForEndpoint = true)
         {
             this.AuthorityUri = authorityUri;
             this.ResourceUri = resourceUri;
             this.identityClientSecret = identityClientSecret;
+            this.clientId = String.IsNullOrEmpty(clientId) ? DefaultIdentityClientId : clientId;
```
Comment (vasont) [inline src/IdentityClient.cs:71]: IsNullOrWhitespace

File: src/IdentityClient.cs [hunk 3]
```diff
@@ -61,11 +63,12 @@ public class IdentityClient
         /// <param name="resourceUri">Contains the URI to the Identity API endpoint.</param>
         /// <param name="identityClientSecret">Contains the API resource client secret.</param>
         /// <param name="useDiscoveryForEndpoint">Contains a value indicating whether the discovery endpoint is used for token endpoint information.</param>
-        public IdentityClient(Uri authorityUri, Uri resourceUri, string identityClientSecret, bool useDiscoveryForEndpoint = true)
+        public IdentityClient(Uri authorityUri, Uri resourceUri, string identityClientSecret, string clientId, bool useDiscoveryForEndpoint = true)
```
Comment (deevezina) [inline src/IdentityClient.cs:66]: Add doc param for clientId

File: src/IdentityClient.cs (hunk 1 above)
Comment (zeroadam) [reply src/IdentityClient.cs:48]: added

File: src/IdentityClient.cs (hunk 2 above)
Comment (zeroadam) [reply src/IdentityClient.cs]: done

**Pull request merged**